mlos_bench/mlos_bench/tests/environments/local_env_test.py

```diff
@@ -66,3 +68,159 @@ def test_local_env(tunable_groups: TunableGroups) -> None:
+    ts2 = ts1 + timedelta(minutes=1)
+
+    time_str1 = ts1.strftime("%Y-%m-%d %H:%M:%S")
+    time_str2 = ts2.strftime("%Y-%m-%d %H:%M:%S")
+
+    local_env = LocalEnv(
+        name="Test Local Env",
+        config={
+            "run": [
+                f"echo {time_str1},cpu_load,0.65 > telemetry.csv",
+                f"echo {time_str1},mem_usage,10240 >> telemetry.csv",
+                f"echo {time_str2},cpu_load,0.8 >> telemetry.csv",
+                f"echo {time_str2},mem_usage,20480 >> telemetry.csv",
+            ],
+            "read_telemetry_file": "telemetry.csv",
+        },
+        service=LocalExecService(parent=ConfigPersistenceService()),
+    )
+
+    with local_env as env_context:
+
+        assert env_context.setup(tunable_groups)
+        (status, _data) = env_context.run()
+        assert status.is_succeeded()
+
+        (status, telemetry) = env_context.status()
+        assert status.is_good()
+        assert telemetry == [
+            (ts1, "cpu_load", 0.65),
+            (ts1, "mem_usage", 10240.0),
+            (ts2, "cpu_load", 0.8),
+            (ts2, "mem_usage", 20480.0),
+        ]
+
+
+def test_local_env_telemetry_invalid(tunable_groups: TunableGroups) -> None:
+    """
+    Fail when the telemetry data has wrong format.
+    """
+    ts1 = datetime.utcnow()
+    ts1 -= timedelta(microseconds=ts1.microsecond)  # Round to a second
+    ts2 = ts1 + timedelta(minutes=1)
+
+    time_str1 = ts1.strftime("%Y-%m-%d %H:%M:%S")
+    time_str2 = ts2.strftime("%Y-%m-%d %H:%M:%S")
+
+    local_env = LocalEnv(
+        name="Test Local Env",
+        config={
+            "run": [
+                # Error: too many columns
+                f"echo {time_str1},EXTRA,cpu_load,0.65 > telemetry.csv",
+                f"echo {time_str1},EXTRA,mem_usage,10240 >> telemetry.csv",
+                f"echo {time_str2},EXTRA,cpu_load,0.8 >> telemetry.csv",
+                f"echo {time_str2},EXTRA,mem_usage,20480 >> telemetry.csv",
+            ],
+            "read_telemetry_file": "telemetry.csv",
+        },
+        service=LocalExecService(parent=ConfigPersistenceService()),
+    )
+
+    with local_env as env_context:
+
+        assert env_context.setup(tunable_groups)
+        (status, _data) = env_context.run()
+        assert status.is_succeeded()
+
+        with pytest.raises(ValueError):
+            env_context.status()
+
+
+def test_local_env_telemetry_invalid_ts(tunable_groups: TunableGroups) -> None:
+    """
+    Fail when the telemetry data has wrong format.
+    """
+    local_env = LocalEnv(
+        name="Test Local Env",
+        config={
+            "run": [
+                # Error: field 1 must be a timestamp
+                "echo 1,cpu_load,0.65 > telemetry.csv",
+                "echo 2,mem_usage,10240 >> telemetry.csv",
+                "echo 3,cpu_load,0.8 >> telemetry.csv",
+                "echo 4,mem_usage,20480 >> telemetry.csv",
```

Comment:
Couldn't these be listed as unix epoch based timestamps (i.e., just ints)?

Reply:
pandas does not recognize an epoch-based int as a timestamp - and I don't want to overengineer. Let's just assume that if pandas cannot automatically infer a datetime, it's an error (btw pandas actually does a really good job recognizing all possible timestamp formats)

Reply:
benchbase outputs unix timestamps with microsecond precision, fwiw